src/gs-category-page.c

```diff
@@ -230,6 +235,82 @@ copy_os_to_usb_button_cb (GtkButton *button, GsCategoryPage *self)
 			 self->cancellable);
 }
 
+static void
+gs_category_page_os_get_copyable_cb (GObject *source_object,
+				     GAsyncResult *res,
+				     gpointer user_data)
+{
+	GsPluginLoader *plugin_loader = GS_PLUGIN_LOADER (source_object);
+	GsCategoryPage *self = GS_CATEGORY_PAGE (user_data);
+	g_autoptr(GError) error = NULL;
+	gboolean copyable;
+	const char *copy_button_label;
+
+	copyable = gs_plugin_loader_job_os_get_copyable_finish (plugin_loader,
+								res, &error);
+	copy_button_label = (copyable ? _("Copy OS to USB") : _("Insert USB to copy"));
```

Comment:
The way it's written now doesn't distinguish between the "Insert USB to copy" and "Copying is impossible" cases. Why set the label just to then make the button invisible?